From ba247fdfd615b00db6fc9b60bb8b3f963f793f17 Mon Sep 17 00:00:00 2001
From: Adrian Hunter <adrian.hunter@intel.com>
Date: Wed, 2 Apr 2014 13:54:59 +0200
Subject: [PATCH] mmc: sdhci: Set the target to present if it is non removable

Some BIOSes misses setting the SDHCI present state register.
Rely on the MMC capabilities to set the present state instead.

This brings WiFi back to life on the Asus T100 after updating
the BIOS to v304.

Change-Id: Ieecb771790d9987bca3ad4b0c60f9fb61d18df5e
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/mmc/host/sdhci.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- linux.orig/drivers/mmc/host/sdhci.c
+++ linux/drivers/mmc/host/sdhci.c
@@ -1343,7 +1343,11 @@ static void sdhci_request(struct mmc_hos
 
 	sdhci_runtime_pm_get(host);
 
-	present = mmc_gpio_get_cd(host->mmc);
+	if (host->mmc->caps & MMC_CAP_NONREMOVABLE)
+		present = 1;
+	else
+		present = mmc_gpio_get_cd(host->mmc);
+
 
 	spin_lock_irqsave(&host->lock, flags);
 
